src/clients/java/src/client.zig

```diff
@@ -93,415 +378,305 @@ const ReflectionHelper = struct {
-        if (result_len > 0) {
-            switch (packet_status) {
-                .ok => if (result_ptr) |ptr| {
-                    // Copying the reply before releasing the packet.
-                    ReflectionHelper.set_reply_buffer(
-                        env,
-                        request_obj,
-                        ptr[0..@intCast(usize, result_len)],
-                    );
-                },
-                else => {},
-            }
-        }
-        context.release_packet(packet);
-
-        ReflectionHelper.end_request(
+    pub inline fn find_field(
+        env: *jni.JNIEnv,
+        class: jni.JClass,
+        comptime name: [:0]const u8,
+        comptime signature: [:0]const u8,
+    ) jni.JFieldID {
+        return env.get_field_id(class, name.ptr, signature.ptr) orelse
+            vm_panic(
             env,
-            request_obj,
-            packet_operation,
-            packet_status,
+            "Field could not be found {s} {s}",
+            .{ name, signature },
         );
     }
-};
-
-/// Export function using the JNI calling convention.
-const Exports = struct {
-    pub fn on_load(vm: *jui.JavaVM) callconv(.C) jui.jint {
-        return NativeClient.on_load(vm);
-    }
-
-    pub fn on_unload(vm: *jui.JavaVM) callconv(.C) void {
-        NativeClient.on_unload(vm);
-    }
 
-    pub fn client_init(
-        env: *jui.JNIEnv,
-        class: jui.jclass,
-        cluster_id: jui.jint,
-        addresses: jui.jstring,
-        max_concurrency: jui.jint,
-    ) callconv(.C) jui.jlong {
-        _ = class;
-        var context = NativeClient.client_init(
-            false,
+    pub inline fn find_method(
+        env: *jni.JNIEnv,
+        class: jni.JClass,
+        comptime name: [:0]const u8,
+        comptime signature: [:0]const u8,
+    ) jni.JMethodID {
+        return env.get_method_id(class, name.ptr, signature.ptr) orelse
+            vm_panic(
             env,
-            @bitCast(u32, cluster_id),
-            addresses,
-            @bitCast(u32, max_concurrency),
+            "Method could not be found {s} {s}",
+            .{ name, signature },
         );
-        return @bitCast(jui.jlong, @ptrToInt(context));
     }
 
-    pub fn client_init_echo(
-        env: *jui.JNIEnv,
-        class: jui.jclass,
-        cluster_id: jui.jint,
-        addresses: jui.jstring,
-        max_concurrency: jui.jint,
-    ) callconv(.C) jui.jlong {
-        _ = class;
-        var context = NativeClient.client_init(
-            true,
-            env,
-            @bitCast(u32, cluster_id),
-            addresses,
-            @bitCast(u32, max_concurrency),
-        );
-        return @bitCast(jui.jlong, @ptrToInt(context));
-    }
+    pub inline fn get_direct_buffer(
+        env: *jni.JNIEnv,
+        buffer_obj: jni.JObject,
+    ) ?[]u8 {
+        var buffer_capacity = env.get_direct_buffer_capacity(buffer_obj);
+        if (buffer_capacity < 0)
+            return null;
```

Comment:
```suggestion
        if (buffer_capacity < 0) return null;
```

let's maybe format on a single line? Multiline ifs without braces are prone to "goto fail"-style bugs (we use zigfmt, so they are unlikely, but still)